### PR TITLE
add OCaml starter with .mli files

### PR DESCRIPTION
**Pull request by smiley1983**
Hi, sorry to close the other request and open a new one. I messed up the git history and reset it the wrong way (using reset rather than revert - still learning).

I've added the .mli files as requested. They're automatically generated, with some function prototypes deleted and some edited to make them shorter and clearer. I left most of the functions in game.mli even though they're not strictly needed - everything left there is of interest to the player, but I could thin it down to just the functions mentioned in the "how to create a starter" documentation.

If this is too much effort, it's no great loss to leave it - historically there hasn't been a lot of interest in using OCaml for these contests, and the ones who do often want to write their own starter rather than using mine. :)

### CODE REVIEW

**Comment by smiley1983** [comment]
Pointing back to development repository for a better view of the latest round of changes:
[halite2-ocaml-starter](https://github.com/smiley1983/halite2-ocaml-starter)

File: airesources/OCaml/game.ml [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+  set_planets_docked_ships state;
+;;
+
+let init state name initial_bot_setup =
+  state.my_id <- int_of_string (Networking.get_string ());
+  Debug.setup_logging state.my_id name;
+  Networking.read_width_height state (Networking.get_string ());
+  update_map state;
+  initial_bot_setup state;
+  Networking.send_string name;
```
Comment (Snaar) [inline airesources/OCaml/game.ml]: I'm still a bit confused about this. This line sends bot name to the server right?

File: airesources/OCaml/game.ml [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+;;
+
+let rec positive_int_degrees a =
+  if a < 0 then positive_int_degrees (a + 360)
+  else a
+;;
+
+let cmd_thrust ship m a =
+  let magnitude = int_of_float m in
+  let angle = positive_int_degrees ((int_of_float (floor (a +. 0.5))) mod 360) in
```
Comment (Snaar) [inline airesources/OCaml/game.ml]: This is not going to round in the right direction with negative angles, right?

File: airesources/OCaml/game.ml [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+let planet_is_owned planet =
+  planet.p_entity.owner >= 0
+;;
+
+let planet_is_full planet =
+  List.length planet.docked_ship_ids >= planet.num_docking_spots
+;;
+
+let rec positive_int_degrees a =
+  if a < 0 then positive_int_degrees (a + 360)
```
Comment (Snaar) [inline airesources/OCaml/game.ml]: You can do this without a conditional: other kits with no modulo operator do it via ((x % 360) + 360) % 360.

File: airesources/OCaml/const.ml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+(* OCaml Starter for Halite II on Halite.io
+   This code is public domain. There is no warranty.
+ *)
+
+let ship_radius = 0.5;;
+
+let dock_radius = 4.;;
+
+let max_speed = 7.;;
```
Comment (Snaar) [inline airesources/OCaml/const.ml]: I might prefer this to be an int, not float, to make it clear speeds as accepted by game are integers.

File: airesources/OCaml/game.ml [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+      start.x *. eend.x -. 
+      start.x *. circle.x +. 
+      eend.x *. circle.x +. 
+     start.y ** 2. -. 
+     start.y *. eend.y -. 
+     start.y *. circle.y +. 
+     eend.y *. circle.y) 
+  in
+(* This variable seems to be unused. *)
+(*  let c = (start.x -. circle.x) ** 2. +. (start.y -. circle.y) ** 2. in *)
```
Comment (Snaar) [inline airesources/OCaml/game.ml]: Yeah, just remove it.

File: airesources/OCaml/game.ml [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+    && intersect_segment_circle e1 e2 obst (Const.ship_radius +. 0.1)
+  ) obstacles
+;;
+
+let direct_navigate ship distance max_speed angle =
+  let speed = if distance >= max_speed then max_speed else distance in
+    (* Not sure why adding 180 is necessary here, but it is. *)
+    (* Perhaps a disagreement about how radians are converted to degrees? *)
+    (* Or where the zero point is? *)
+    cmd_thrust ship speed (angle +. 180.)
```
Comment (Snaar) [inline airesources/OCaml/game.ml]: This makes me uncomfortable :-)
Why do you think it's necessary here?

Comment (smiley1983) [reply airesources/OCaml/game.ml]: Now fixed. I had imported my rad2deg and deg2rad functions from another project without checking them properly. No more adding of 180 degrees.

I've also made the other changes suggested.

**Comment by harikmenon** [comment]
@Snaar are we good to go with this bot?

**Comment by Snaar** [comment]
Yeah, sorry, was out for few days. Looks good now.

**Pull request merged**